### DISCUSSION TITLE
JSX Layer

### DISCUSSION THREAD

**Discussion by floki2020** [visgl/deck.gl#6383] [Q&A]
I use the demo of website about Use JSX Layer  like this
```
<DeckGL initialViewState={initView} layers={layers} controller={true} onClick={clickMap}>
        <InteractiveMap reuseMaps mapboxApiAccessToken={MAPBOX_TOKEN} mapStyle={STYLE} />
        <LineLayer id={'123'} data={data}/>
      </DeckGL>

```
But it didn't work！ERROR LineLayer is Invalid JSX element！
deck.GL :8.6
react ：17
typescript：4
Thanks

**Comment by Pessimistress** [comment]
JSX layers require all the layer classes to be derived from the core `Layer` class. If you are seeing this error it's possible that you have multiple copies of `@deck.gl/core` installed. You can check this by running `yarn list` or `npm ls`.

**Reply by floki2020** [reply]
Thank u for ur replay！！
I used the LineLayer extends from the BASE Layer ！And i check it out！
─ @ant-design/charts@1.2.14
├── @ant-design/icons@4.7.0
├── @ant-design/pro-cli@2.1.5
├── @ant-design/pro-descriptions@1.9.27
├── @ant-design/pro-form@1.45.0        
├── @ant-design/pro-layout@6.27.1      
├── @ant-design/pro-table@2.56.6       
├── @danmarshall/deckgl-typings@4.9.7  
├── @deck.gl/mesh-layers@8.6.3
├── @loaders.gl/core@3.0.12
├── @loaders.gl/gltf@3.0.12
├── @loaders.gl/obj@3.0.13
├── @mapbox/mapbox-gl-language@1.0.0   
├── @nebula.gl/layers@0.23.8
├── @nebula.gl/overlays@0.23.8
├── @types/d3-scale@4.0.2
├── @types/express@4.17.13
├── @types/history@4.7.9
├── @types/jest@26.0.24
├── @types/lodash@4.14.176
├── @types/react-dom@17.0.10
├── @types/react-helmet@6.1.4
├── @types/react@17.0.33
├── @umijs/fabric@2.8.1
├── @umijs/openapi@1.3.0
├── @umijs/plugin-blocks@2.2.2
├── @umijs/plugin-esbuild@1.4.1
├── @umijs/plugin-openapi@1.3.1
├── @umijs/preset-ant-design-pro@1.3.3
├── @umijs/preset-dumi@1.1.31
├── @umijs/preset-react@1.8.25
├── @umijs/route-utils@1.0.37
├── @umijs/yorkie@2.0.5
├── @vime/core@5.3.0
├── @vime/react@5.3.0
├── antd@4.16.13
├── carlo@0.9.46
├── classnames@2.3.1
├── cross-env@7.0.3
├── cross-port-killer@1.3.0
├── deck.gl@8.6.0
├── detect-installer@1.0.2
├── echarts@5.2.2
├── enzyme@3.11.0
├── eslint@7.32.0
├── express@4.17.1
├── file-loader@6.2.0
├── gh-pages@3.2.3
├── jsdom-global@3.0.2
├── lint-staged@10.5.4
├── lodash@4.17.21
├── mapbox-gl@2.5.1
├── mockjs@1.1.0
├── moment@2.29.1
├── omit.js@2.0.2
├── prettier@2.4.1
├── puppeteer-core@8.0.0
├── rc-menu@9.0.14
├── rc-util@5.14.0
├── react-dev-inspector@1.7.0
├── react-dom@17.0.2
├── react-helmet-async@1.1.2
├── react-map-gl@6.1.17
├── react@17.0.2
├── stylelint@13.13.1
├── swagger-ui-react@3.52.5
├── typescript@4.4.4
├── umi-serve@1.9.10
└── umi@3.5.20

so..
